### PR TITLE
Added Consul Task

### PR DESCRIPTION
**Pull request by GregHanson**
steps for running consul on docker for the website

### CODE REVIEW

**Comment by ldemailly** [comment]
```
  *  External link https://github.com/istio/istio.github.io/edit/master/_docs/tasks/consul.md failed: 404 No error
```
is expected for new files, the others not so much (looks like envoy is moving out of lyft web site wise)

File: _docs/tasks/consul.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+overview: Make Istio run in docker environment by integrating Consul as a service registry.
+
+order: 42
+
+layout: docs
+type: markdown
+---
+
+Configure Istio to run in docker environment by integrating Consul as a service registry.
+
```
Comment (linsun) [inline _docs/tasks/consul.md]: docker -> Docker

File: _docs/tasks/consul.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+## Before you begin
+
+* Setup Istio by following the instructions in the [Installation guide](./installing-istio.html).  You will
+  need `kubectl` and Steps 1-4 from `Installation step` only for this Task.
+
```
Comment (linsun) [inline _docs/tasks/consul.md]: should we say `kubectl` 1.7.3 or later?

File: _docs/tasks/consul.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+## Overview
+
+This sample deploys the [BookInfo]({{home}}/docs/samples/bookinfo.html) application. The Consul platform 
+adapter uses Consul Server to help Istio monitor service instances running in the underlying platform. 
+When a service instance is brought up in docker, the [Registrator](http://gliderlabs.github.io/registrator/latest/) 
+automatically registers the service in Consul.
+
+Note that Istio pilot is running inside each app container so as to coordinate Envoy and the service mesh.
+
```
Comment (linsun) [inline _docs/tasks/consul.md]: should we say Istio pilot agent here?  I think Istio pilot is confusing as not entire pilot runs inside of each app container, correct?

File: _docs/tasks/consul.md [hunk 4]
```diff
@@ -0,0 +1,101 @@
+---
+title: Running Istio in Docker with Consul
+overview: Make Istio run in docker environment by integrating Consul as a service registry.
+
+order: 42
```
Comment (rshriram) [inline _docs/tasks/consul.md]: Do you have a screenshot showing how this appears ?

File: _docs/tasks/consul.md (hunk 2 above)
Comment (GregHanson) [reply _docs/tasks/consul.md]: If there is a version dependency, I would hope it would be in the installing-istio.html page - right?

Comment (linsun) [reply _docs/tasks/consul.md]: ok that sounds good.

File: _docs/tasks/consul.md (hunk 4 above)
Comment (GregHanson) [reply _docs/tasks/consul.md]: I ran the following steps on my mac from the README:
```
$ xcode-select --install
$ brew install ruby
$ sudo gem install bundler
$ sudo gem install jekyll
$ cd istio.github.io
$ bundle install
$ bundle exec jekyll build
$ bundle exec jekyll serve
```
went to the Tasks page, made sure it was included in the list, and that the link went to my new page

Comment (GregHanson) [reply _docs/tasks/consul.md]: screen shot 2017-09-15 at 2 36 02 pm

Comment (GregHanson) [reply _docs/tasks/consul.md]: screen shot 2017-09-15 at 2 37 12 pm

Comment (ldemailly) [reply _docs/tasks/consul.md]: seems like we should tweak the order? (priority setting)

**Comment by ldemailly** [comment]
*  External link https://github.com/istio/istio.github.io/edit/master/_docs/tasks/consul.md failed: 404 No error
is expected so once approved this can go in

File: _docs/tasks/consul.md (hunk 4 above)
Comment (GregHanson) [reply _docs/tasks/consul.md]: @ldemailly do you mean the order for my new item? or the order in general?

Comment (rshriram) [reply _docs/tasks/consul.md]: Can you change the Installing Istio to Installing Istio in Kubernetes, and then change this to Installing Istio in Docker Swarm w/ Consul

Comment (rshriram) [reply _docs/tasks/consul.md]: I also expect a follow up PR in the Integrating services into the mesh, where you need to have a special section for plain docker environments.

**Comment by ldemailly** [comment]
this broke the main page / please fix
(there should be an error about in check too)

**Comment by ldemailly** [comment]
I reverted for now - please re-submit with the following caveat:

a) do not rewrite history, changing one of our already posted blog post is a smoking gun that something was not right (a bunch of sites, blogs, ... link into istio install - do not break that url

b) do not force merge with obvious errors without checking the presubmit logs and/or make further change after review comment (#453)

please let me review the next version and other significant structural changes before they are merged


proposal : make installing istio a task with sub tasks or links to the consul one

@andraxylia is slated to rewrite the k8s install too btw so coordinate with her

**Comment by andraxylia** [comment]
Yes, please, let's keep main install page short with links to the variations.

**Comment by rshriram** [comment]
FYI
https://www.google.com/search?q=https://istio.io/docs/tasks/installing-istio.html&ei=ft2_Wb_9HMaWmQGkjbAI&start=0&sa=N&biw=1276&bih=704

shows all links from our own website.. And if you go to pages 2, 3 - there are just 2 links that point to this specific link. Nothing else. That said, I will still retain the install page and move kubernetes, consul into separate pages.

**Pull request merged**